### PR TITLE
Edit Post: Refactor `KeyboardShortcutHelpModal` tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by tyxla**
## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `KeyboardShortcutHelpModal` tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers and mocks to avoid testing unrelated implementation details.

Ideally, we should not be using snapshot tests here, however, the purpose of this PR is not to improve or enrich the tests themselves, but rather to migrate them `@testing-library/react`. Our primary motivation is unblocking the upgrade to React 18.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/edit-post/src/components/keyboard-shortcut-help-modal/test/index.js`

### CODE REVIEW

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap [hunk 1]
```diff
@@ -1,106 +1,381 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`KeyboardShortcutHelpModal should match snapshot when the modal is active 1`] = `
-<ForwardRef(Modal)
-  className="edit-post-keyboard-shortcut-help-modal"
-  closeLabel="Close"
```
Comment (ciampo) [inline packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap:6]: Just noting that in the old snapshot, `closeLabel` had a value of `Close`, but in the new snapshot the label of the close button seems to be `Close dialog`. Perhaps it's a bug that we should look at?

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -1,106 +1,381 @@
...
+      class="edit-post-keyboard-shortcut-help-modal__section edit-post-keyboard-shortcut-help-modal__main-shortcuts"
+    >
+      <ul
+        class="edit-post-keyboard-shortcut-help-modal__shortcut-list"
+        role="list"
+      >
+        <li
+          class="edit-post-keyboard-shortcut-help-modal__shortcut"
+        />
+      </ul>
```
Comment (ciampo) [inline packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap:56]: Is a list with just one (empty) `li` correct here? This doesn't look like correct markup

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap [hunk 3]
```diff
@@ -1,106 +1,381 @@
...
+    >
+      <h2
+        class="edit-post-keyboard-shortcut-help-modal__section-title"
+      >
+        Global shortcuts
+      </h2>
+      <ul
+        class="edit-post-keyboard-shortcut-help-modal__shortcut-list"
+        role="list"
+      />
```
Comment (ciampo) [inline packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap]: Another empty list, which doesn't look like correct markup

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap [hunk 4]
```diff
@@ -1,106 +1,381 @@
...
+    >
+      <h2
+        class="edit-post-keyboard-shortcut-help-modal__section-title"
+      >
+        Selection shortcuts
+      </h2>
+      <ul
+        class="edit-post-keyboard-shortcut-help-modal__shortcut-list"
+        role="list"
+      />
```
Comment (ciampo) [inline packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap]: Same

**Review by ciampo — COMMENTED** [review]
The refactor in itself looks good, but the new snapshot may have highlighted a few problems with the generated markup.

We should look into this and assess if the component experienced a few regressions over time before merging (or if maybe it's a matter of mocking some data APIs)?

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap (hunk 1 above)
Comment (tyxla) [reply packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap:6]: Oh, that's a great find! 💯 

Seems like it was caused by the wrong close button label prop used in the original component. 🐛 

I've fixed that in 68aaa0e7971ffd24f6ef4378ebabdc2aaa2bafea.

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap (hunk 2 above)
Comment (tyxla) [reply packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap:56]: Seems like we also had to register the editor shortcuts. That is as simple as rendering an `<EditorKeyboardShortcutsRegister />` instance in our test. Done in 4e5e1a0c7fd8aadf12b03357b97f948a8bdd92bb.

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap (hunk 3 above)
Comment (tyxla) [reply packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap]: Done as part of 4e5e1a0c7fd8aadf12b03357b97f948a8bdd92bb.

File: packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap (hunk 4 above)
Comment (tyxla) [reply packages/edit-post/src/components/keyboard-shortcut-help-modal/test/__snapshots__/index.js.snap]: Fixed as part of 4e5e1a0c7fd8aadf12b03357b97f948a8bdd92bb.

**Comment by tyxla** [comment]
Thank you for the great catches, @ciampo! 

I've addressed them and this PR is ready for another look 👀 when you get a chance.

**Comment by ciampo** [comment]
Thank you for fixing the bug!

There are some e2e failures that seem related — it could be due to the fact that the close button's label has changed?

**Comment by tyxla** [comment]
> Thank you for fixing the bug!
> 
> There are some e2e failures that seem related — it could be due to the fact that the close button's label has changed?

Oh, indeed. Seems like we need to update the e2e test - done in https://github.com/WordPress/gutenberg/pull/44077/commits/aa40e4ffba4d009b7de2fbb4cb6ac1cba0bcde4a

**Comment by tyxla** [comment]
Could I get another look when you get a chance @ciampo?

**Review by ciampo — APPROVED** [review]
🚀

**Pull request merged**